File: split.js
const fs = require('fs');

const content = fs.readFileSync('index.html', 'utf8');

// Extract parts
const headPattern = /(<!DOCTYPE html>[\s\S]*?<body>)/;
const head = content.match(headPattern)[1];

const headerPattern = /(<!-- Background Decorative Blobs -->[\s\S]*?<\/header>)/;
const header = content.match(headerPattern)[1];

const footerPattern = /(<!-- 7\. RODAPÉ \(FOOTER\) -->[\s\S]*?<\/html>)/;
const footer = content.match(footerPattern)[1];

function updateNav(html) {
    let result = html;
    result = result.replace(/href="#inicio"/g, 'href="index.html#inicio"');
    result = result.replace(/href="#sobre"/g, 'href="sobre.html#sobre"');
    result = result.replace(/href="#especialidades"/g, 'href="sobre.html#especialidades"');
    result = result.replace(/href="#metodologia"/g, 'href="index.html#metodologia"');
    result = result.replace(/href="#videos"/g, 'href="avaliacoes.html#videos"');
    result = result.replace(/href="#jornada"/g, 'href="index.html#jornada"');
    result = result.replace(/href="#depoimentos"/g, 'href="avaliacoes.html#depoimentos"');
    result = result.replace(/href="#faq"/g, 'href="index.html#faq"');
    result = result.replace(/href="#contato"/g, 'href="index.html#contato"');
    return result;
}

const updatedHeader = updateNav(header);

const heroMatch = content.match(/(<!-- 2\. SEÇÃO HERO[\s\S]*?)<\/section>/);
const hero = heroMatch[1] + '</section>';

const sobreMatch = content.match(/(<!-- 3\. SEÇÃO "SOBRE MIM"[\s\S]*?)<\/section>/);
const sobre = sobreMatch[1] + '</section>';

const especialidadesMatch = content.match(/(<!-- 4\. SEÇÃO "ÁREAS DE ATUAÇÃO \/ ESPECIALIDADES"[\s\S]*?)<\/section>/);
const especialidades = especialidadesMatch[1] + '</section>';

const metodologiaMatch = content.match(/(<!-- 5\. SEÇÃO "METODOLOGIA: POR QUE É DIFERENTE\?"[\s\S]*?)<\/section>/);
const metodologia = metodologiaMatch[1] + '</section>';

const videosMatch = content.match(/(<!-- SEÇÃO VÍDEOS & CONTEÚDOS DO YOUTUBE -->[\s\S]*?)<\/section>/);
const videos = videosMatch[1] + '</section>';

const jornadaMatch = content.match(/(<!-- NEW SECTION 1: JORNADA DO PACIENTE -->[\s\S]*?)<\/section>/);
const jornada = jornadaMatch[1] + '</section>';

const depoimentosMatch = content.match(/(<!-- SEÇÃO DEPOIMENTOS \/ AVALIAÇÕES DO GOOGLE -->[\s\S]*?<div class="carousel-action text-center mt-lg fade-in-up">[\s\S]*?<\/div>[\s\S]*?<\/div>[\s\S]*?)<\/section>/);
const depoimentos = depoimentosMatch[1] + '</section>';

const faqMatch = content.match(/(<!-- NEW SECTION 3: FAQ \/ PERGUNTAS FREQUENTES -->[\s\S]*?)<\/section>/);
const faq = faqMatch[1] + '</section>';

const locationMatch = content.match(/(<!-- NEW SECTION 4: LOCALIZAÇÃO & MODALIDADES DE ATENDIMENTO -->[\s\S]*?)<\/section>/);
const location = locationMatch[1] + '</section>';

const contactMatch = content.match(/(<!-- 6\. SEÇÃO DE AGENDAMENTO E CONTATO -->[\s\S]*?)<\/section>/);
const contact = contactMatch[1] + '</section>';

const footerCleanMatch = content.match(/(<!-- 7\. RODAPÉ \(FOOTER\) -->[\s\S]*?<\/footer>)/);
const footerClean = updateNav(footerCleanMatch[1]);

const whatsappMatch = content.match(/(<!-- Floating WhatsApp Widget -->[\s\S]*?<\/a>)/);
const whatsapp = whatsappMatch[1];

const modalMatch = content.match(/(<!-- Review Modal Overlay -->[\s\S]*?<\/div>\s*<\/div>)/);
const modal = modalMatch[1];

const scripts = '\n    <script src="script.js"></script>\n</body>\n</html>';

let newIndex = `${head}\n${updatedHeader}\n${hero}\n${metodologia}\n${jornada}\n${faq}\n${location}\n${contact}\n${footerClean}\n${whatsapp}\n${scripts}`;

let newSobre = `${head}\n${updatedHeader}\n${sobre}\n${especialidades}\n${footerClean}\n${whatsapp}\n${scripts}`;
newSobre = newSobre.replace('<title>Helena Ávila | Psicoterapeuta e Psicanalista</title>', '<title>Sobre | Helena Ávila</title>');

let newAvaliacoes = `${head}\n${updatedHeader}\n${videos}\n${depoimentos}\n${footerClean}\n${whatsapp}\n${modal}\n${scripts}`;
newAvaliacoes = newAvaliacoes.replace('<title>Helena Ávila | Psicoterapeuta e Psicanalista</title>', '<title>Avaliações | Helena Ávila</title>');

fs.writeFileSync('index.html', newIndex, 'utf8');
fs.writeFileSync('sobre.html', newSobre, 'utf8');
fs.writeFileSync('avaliacoes.html', newAvaliacoes, 'utf8');

console.log("Split completed successfully!");

File: split.py
import re
import os

with open('index.html', 'r', encoding='utf-8') as f:
    content = f.read()

# Extract parts
head_pattern = re.compile(r'(<!DOCTYPE html>.*?<body>)', re.DOTALL)
head = head_pattern.search(content).group(1)

header_pattern = re.compile(r'(<!-- Background Decorative Blobs -->.*?</header>)', re.DOTALL)
header = header_pattern.search(content).group(1)

footer_pattern = re.compile(r'(<!-- 7\. RODAPÉ \(FOOTER\) -->.*?</html>)', re.DOTALL)
footer = footer_pattern.search(content).group(1)

# Fix navigation links in header and footer
def update_nav(html):
    html = html.replace('href="#inicio"', 'href="index.html#inicio"')
    html = html.replace('href="#sobre"', 'href="sobre.html#sobre"')
    html = html.replace('href="#especialidades"', 'href="sobre.html#especialidades"')
    html = html.replace('href="#metodologia"', 'href="index.html#metodologia"')
    html = html.replace('href="#videos"', 'href="avaliacoes.html#videos"')
    html = html.replace('href="#jornada"', 'href="index.html#jornada"')
    html = html.replace('href="#depoimentos"', 'href="avaliacoes.html#depoimentos"')
    html = html.replace('href="#faq"', 'href="index.html#faq"')
    html = html.replace('href="#contato"', 'href="index.html#contato"')
    return html

header = update_nav(header)
footer = update_nav(footer)

# Extract sections
hero = re.search(r'(<!-- 2\. SEÇÃO HERO.*?)</section>', content, re.DOTALL).group(1) + '</section>'
sobre = re.search(r'(<!-- 3\. SEÇÃO "SOBRE MIM".*?)</section>', content, re.DOTALL).group(1) + '</section>'
especialidades = re.search(r'(<!-- 4\. SEÇÃO "ÁREAS DE ATUAÇÃO / ESPECIALIDADES".*?)</section>', content, re.DOTALL).group(1) + '</section>'
metodologia = re.search(r'(<!-- 5\. SEÇÃO "METODOLOGIA: POR QUE É DIFERENTE\?".*?)</section>', content, re.DOTALL).group(1) + '</section>'
videos = re.search(r'(<!-- SEÇÃO VÍDEOS & CONTEÚDOS DO YOUTUBE -->.*?)</section>', content, re.DOTALL).group(1) + '</section>'
jornada = re.search(r'(<!-- NEW SECTION 1: JORNADA DO PACIENTE -->.*?)</section>', content, re.DOTALL).group(1) + '</section>'
depoimentos = re.search(r'(<!-- SEÇÃO DEPOIMENTOS / AVALIAÇÕES DO GOOGLE -->.*?<div class="carousel-action text-center mt-lg fade-in-up">.*?</div>.*?</div>.*?)</section>', content, re.DOTALL).group(1) + '</section>'
faq = re.search(r'(<!-- NEW SECTION 3: FAQ / PERGUNTAS FREQUENTES -->.*?)</section>', content, re.DOTALL).group(1) + '</section>'
location = re.search(r'(<!-- NEW SECTION 4: LOCALIZAÇÃO & MODALIDADES DE ATENDIMENTO -->.*?)</section>', content, re.DOTALL).group(1) + '</section>'
contact = re.search(r'(<!-- 6\. SEÇÃO DE AGENDAMENTO E CONTATO -->.*?)</section>', content, re.DOTALL).group(1) + '</section>'

# Create new index.html
new_index = f"{head}\n{header}\n{hero}\n{metodologia}\n{jornada}\n{faq}\n{location}\n{contact}\n{footer}"

# Create sobre.html
new_sobre = f"{head}\n{header}\n{sobre}\n{especialidades}\n{footer}"
new_sobre = new_sobre.replace('<title>Helena Ávila | Psicoterapeuta e Psicanalista</title>', '<title>Sobre | Helena Ávila</title>')

# Create avaliacoes.html
# Need the review modal for avaliacoes.html
modal = re.search(r'(<!-- Review Modal Overlay -->.*?</div>\s*</div>)', content, re.DOTALL).group(1)
new_avaliacoes = f"{head}\n{header}\n{videos}\n{depoimentos}\n{footer}"
new_avaliacoes = new_avaliacoes.replace('<!-- Review Modal Overlay -->', modal + '\n\n    <!-- Review Modal Overlay -->') # Ensure modal is there
# Wait, footer already includes the modal and floating whatsapp because it matches from footer to </html>
# Ah! Let's check where the modal is. It's inside the footer match if it goes to </html>.
# Let's clean up the modal logic.
footer_clean = re.search(r'(<!-- 7\. RODAPÉ \(FOOTER\) -->.*?</footer>)', content, re.DOTALL).group(1)
whatsapp = re.search(r'(<!-- Floating WhatsApp Widget -->.*?</a>)', content, re.DOTALL).group(1)
scripts = '\n    <script src="script.js"></script>\n</body>\n</html>'

new_index = f"{head}\n{header}\n{hero}\n{metodologia}\n{jornada}\n{faq}\n{location}\n{contact}\n{footer_clean}\n{whatsapp}\n{scripts}"

new_sobre = f"{head}\n{header}\n{sobre}\n{especialidades}\n{footer_clean}\n{whatsapp}\n{scripts}"
new_sobre = new_sobre.replace('<title>Helena Ávila | Psicoterapeuta e Psicanalista</title>', '<title>Sobre | Helena Ávila</title>')

new_avaliacoes = f"{head}\n{header}\n{videos}\n{depoimentos}\n{footer_clean}\n{whatsapp}\n{modal}\n{scripts}"
new_avaliacoes = new_avaliacoes.replace('<title>Helena Ávila | Psicoterapeuta e Psicanalista</title>', '<title>Avaliações | Helena Ávila</title>')

new_index = update_nav(new_index)
new_sobre = update_nav(new_sobre)
new_avaliacoes = update_nav(new_avaliacoes)

with open('index.html', 'w', encoding='utf-8') as f:
    f.write(new_index)

with open('sobre.html', 'w', encoding='utf-8') as f:
    f.write(new_sobre)

with open('avaliacoes.html', 'w', encoding='utf-8') as f:
    f.write(new_avaliacoes)

print("Split completed successfully!")
